tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PagingSendTest.java

```diff
@@ -288,6 +296,408 @@ public void testPagingDoesNotDuplicateBatchMessagesAfterPagingStarted() throws E
+
+      final String queueName = getTestMethodName();
+
+      try (ClientSessionFactory sf = createSessionFactory(locator)) {
+         ClientSession session = sf.createSession(true, true);
+
+         SimpleString queueAddr = SimpleString.of(queueName);
+         session.createQueue(QueueConfiguration.of(queueAddr));
+
+         final int size = 1024 * 50;
+         final Long maxMessages = 10L;
+         AddressSettings addressSettings = new AddressSettings();
+         addressSettings.setAddressFullMessagePolicy(AddressFullMessagePolicy.PAGE);
+         addressSettings.setPageFullMessagePolicy(PageFullMessagePolicy.FAIL);
+         addressSettings.setPageSizeBytes(size);
+         addressSettings.setPageLimitBytes((long) (size * 10));
+         addressSettings.setMaxSizeBytes(size);
+         addressSettings.setPageLimitMessages(maxMessages);
+
+         server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+         int totalMessages = 0;
+         int messageSize = 1024;
+         ClientProducer producer = session.createProducer(queueAddr);
+         boolean stop = false;
+         while (!stop) {
+            try {
+               ClientMessage message = createMessage(session, messageSize, totalMessages, null);
+               producer.send(message);
+               totalMessages++;
+               session.commit();
+            } catch (ActiveMQAddressFullException ex) {
+               stop = true;
+            }
+         }
+
+         Queue queue = server.locateQueue(queueAddr);
+         assertTrue(waitForMessages(queue, totalMessages, 10000));
+
+         PagingStore queuePagingStore = queue.getPagingStore();
+         assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+         // the messages reach the limit
+         PageCursorProvider cursorProvider = queuePagingStore.getCursorProvider();
+         assertEquals(PageCursorProviderTestAccessor.getNumberOfMessagesOnSubscriptions(cursorProvider), maxMessages);
+
+         // but pages still under limit
+         long existingPages = queuePagingStore.getNumberOfPages();
+         assertTrue(existingPages <= queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes());
+
+         server.stop(true);
+         waitForServerToStop(server);
+
+         // restart the server the pageFull is still true
+         try {
+            server.start();
+            waitForServerToStart(server);
+
+            queue = server.locateQueue(queueAddr);
+
+            queuePagingStore = queue.getPagingStore();
+            assertTrue(queuePagingStore != null && queuePagingStore.isPageFull());
+
+            // but current pages still under limit
+            long currentPages = queuePagingStore.getNumberOfPages();
+            long maxPages = queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes();
+            assertTrue(currentPages <= maxPages);
```

Comment:
Similarly here, it should validate against a pre-known value and/or check the getters are correct, rather than trusting that they are.